### PR TITLE
feat: Add task runners

### PR DESCRIPTION
**Pull request by pythoninthegrass**
### Checklist

* [x] I have read the Contribution Guidelines
* [x] I have read and agree to the Code of Conduct
* [x] I have read and agree that my contributions will be included under the project license
* [x] I have added a description of my changes and why I want them _or_ linked the relevant pull request below
* [ ] I have ran the formatter, linter and tests locally and ensured they give their all green
* [ ] If contributing new code: I've ensured my new code is covered sufficiently by unit and integration tests

I've manually tested all of my changes (most were lift and shift from my other projects), but when running `pytest` I'm getting unrelated `OSError` tracebacks. Most likely due to being on an M1 Mac.

Not really relevant to build/ci tooling haha.

### Description of Changes

- `Makefile`: handle bare minimum dependencies
- `taskfile.yml`: entrypoint for miscellaneous tasks and subtasks
- `docker.yml`: future proof later commits with docker
- `poetry.yml`: common tasks incl. `ci.bash` coverage report
- `.env.example`: placeholders for `poetry.yml` pypi env vars
- `README.md`: added missing step to contribution, general task runner steps
- `requirements.txt`: backward compatibility w/pip

### Related Issues

N/A

### CODE REVIEW

File: .env.example
Comment (thijsmie) [inline .env.example:1]: Since this project is released via the github trusted publisher system we don't need this.

File: Makefile
Comment (thijsmie) [inline Makefile:1]: I do like makefiles, but I don't think they're the right tool for the job in a python library project. Some python based dev tooling is more appropriate, such as tox/nox or invoke.

Besides, I see a lot of tooling here that I don't think belongs to this project. I personally disagree with many of the choices the maintainer of pre-commit made, but that aside: asdf, brew, docker and devbox all are up to the developer and not up to the project to prescribe.

File: README.md [hunk 1]
```diff
@@ -55,9 +55,10 @@ Don't forget to give the project a star! Thanks again!
 
 1. Fork the project
 2. Create your feature branch (`git checkout -b feat/some-nice-feature`)
-3. Commit your changes (`git commit -m 'feat: Add some AmazingFeature'`)
-4. Push to the pranch (`git push -u origin feat/some-nice-feature`)
-5. Open a pull request
+3. Add your changes (`git add .`)
```
Comment (thijsmie) [inline README.md:58]: Good catch!

File: README.md [hunk 2]
```diff
@@ -73,6 +74,34 @@ The `pocketbase-async` package was inspired and guided in implementation by seve
 
 Furthermore, a lot of the API tests were adapted from Vaphes' work (licensed MIT).
 
+## Development
```
Comment (thijsmie) [inline README.md:77]: We indeed probably need a section like this in the to be written documentation, maybe not in the readme.md

File: requirements.txt
Comment (thijsmie) [inline requirements.txt:1]: I don't plan to provide the requirements.txt. You don't need it if installing via the registry and if you are developing on the project directly you're expected to take the time to work with poetry.

File: taskfile.yml
Comment (thijsmie) [inline taskfile.yml:1]: Another third-party non-python task runner definition to maintain. You're free to use this in your own environment, its not going into the repository.

File: tasks/docker.yml
Comment (thijsmie) [inline tasks/docker.yml:1]: See above.

File: tasks/poetry.yml
Comment (thijsmie) [inline tasks/poetry.yml:1]: See above.

**Review by thijsmie — COMMENTED** [review]
Hi there @pythoninthegrass!

Thank you for taking the time to prepare this pull request. I really appreciate your willingness to contribute. However, I do think maybe it would have been better to have a quick discussion before you dove into this. I've put some remarks down as comments on the files directly, please excuse my dutch directness.

**Comment by pythoninthegrass** [comment]
Sounds like you've got everything handled. I'll leave you to it

**Pull request closed without merging**